libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/backpress/OnKeyListenerComponent.kt

```diff
@@ -6,25 +6,27 @@ import com.mapbox.navigation.core.MapboxNavigation
 import com.mapbox.navigation.dropin.DropInNavigationView
 import com.mapbox.navigation.dropin.DropInNavigationViewContext
 import com.mapbox.navigation.dropin.component.destination.DestinationAction
-import com.mapbox.navigation.dropin.component.navigationstate.NavigationState
+import com.mapbox.navigation.dropin.component.navigation.NavigationState
+import com.mapbox.navigation.dropin.component.navigation.NavigationStateAction
 import com.mapbox.navigation.dropin.component.routefetch.RoutesAction
 import com.mapbox.navigation.dropin.lifecycle.UIComponent
 
 /**
  * Key listener for the [DropInNavigationView].
  *
- * On back pressed will change the navigation state.
- * (FreeDrive) <- (RoutePreview) <- (ActiveNavigation)
- *                                  (Arrival)
+ * On back pressed will update the navigation state.
+ *
+ * (FreeDrive) <- (DestinationPreview) <- (RoutePreview) <- (ActiveNavigation)
+ *             <- (Arrival)
  */
 internal class OnKeyListenerComponent(
-    context: DropInNavigationViewContext,
+    private val context: DropInNavigationViewContext,
```

Comment:
Instead of `DropInNavigationViewContext` can we inject the viewModels themselves?

Reply:
Moving this conversation to this thread https://github.com/mapbox/mapbox-navigation-android/pull/5594#discussion_r833728678